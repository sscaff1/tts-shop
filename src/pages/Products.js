import React, { useEffect } from 'react';
import { Spin } from 'antd';
import { useDispatch, useSelector } from 'react-redux';
import Product from 'components/products/Product';
import productActions from 'actions/products';

import styles from './Products.module.css';

function Products() {
  const dispatch = useDispatch();
  const { products, ui } = useSelector(state => state);
  useEffect(() => {
    dispatch(productActions.getProducts());
  }, [dispatch]);
  if (ui.loading) {
    return (
      <div className={styles.spin}>
        <Spin size="large" />
      </div>
    );
  }
  return (
    <div className={styles.grid}>
      {products.map(product => (
        <Product
          key={product.id}
          title={product.title}
          description={product.description}
          pictureUrl={product.pictureUrl}
          price={product.price}
          id={product.id}
        />
      ))}
    </div>
  );
}

export default Products;
