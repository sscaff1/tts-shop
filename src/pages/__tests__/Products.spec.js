import React from 'react';
import {
  render,
  fireEvent,
  cleanup,
  waitForElementToBeRemoved,
} from '@testing-library/react';
import { Provider } from 'react-redux';
import Products from '../Products';
import createStore from '../../store';

jest.mock('utils/firebase');

describe('Products Page', () => {
  afterEach(cleanup);
  const setup = () => {
    return (
      <Provider store={createStore()}>
        <Products />
      </Provider>
    );
  };

  it('should start with loading screen', async () => {
    const { container } = render(setup());
    const spinner = container.querySelector('.spin');

    expect(spinner).toBeTruthy();
  });

  it('should render', async () => {
    const { container } = render(setup());

    await waitForElementToBeRemoved(() => container.querySelector('.spin'));

    expect(container).toMatchSnapshot();
  });
});
