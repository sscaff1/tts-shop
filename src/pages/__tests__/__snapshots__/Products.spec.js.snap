// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products Page should render 1`] = `
<div>
  <div
    class="grid"
  >
    <div
      class="ant-card ant-card-bordered ant-card-hoverable"
      style="max-width: 30%; min-width: 30%; flex: 1; margin-bottom: 15px; margin-right: 15px;"
    >
      <div
        class="ant-card-cover"
      >
        <img
          alt="Test"
          src="test picture"
        />
      </div>
      <div
        class="ant-card-body"
      >
        <div
          class="ant-card-meta"
        >
          <div
            class="ant-card-meta-detail"
          >
            <div
              class="ant-card-meta-title"
            >
              Test
            </div>
            <div
              class="ant-card-meta-description"
            >
              test description
            </div>
          </div>
        </div>
        <h4>
          $10.00
        </h4>
        <div>
          <button
            class="ant-btn"
            type="button"
          >
            <span>
              Add to Cart
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
